task/alerts/alert_signal

BarAgam opened the pull request:
Notifying user when a new alerts is received.
App's toolbar is animating from its original color to red (repeatedly) and it's (the toolbar's) title is changing to alert.
The toolbar status is being restored after clicking on it.

gRaytan committed:
starting implementing display bubble alert

gRaytan committed:
Support basic Alert.
1. Alert entity created and its support:
   a. AlertMessage
   b. AlertData
   c. Transformers
2. Alerts repository.
3. Alerts messages are added to the repo.

gRaytan committed:
implementing getting alerts #2

barmako committed:
Notifying user when a new alerts is received.
App's toolbar is animating from its original color to red (repeatedly) and it's (the toolbar's) title is changing to alert.
The toolbar status is being restored after clicking on it.

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java:
```diff
@@ -0,0 +1,131 @@
+
+
+    public BubbleAlertsSubcomponent(Activity activity) {
+        mToolbar = (Toolbar) activity.findViewById(R.id.main_toolbar);
```
ButterKnife can be used here

BarAgam replied:
Unfortunately I couldn't make it happen here.

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java:
```diff
@@ -0,0 +1,131 @@
+        }
+    }
+
+    private class MyTitleSetter implements BubbleAlertsViewModel.TitleSetter {
```
Change the name of the title setter

teamagam3 commented on app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java:
```diff
@@ -0,0 +1,131 @@
+        mBubbleAlertsViewModel.onToolbarClick();
+    }
+
+    private class MyToolbarAnimator implements BubbleAlertsViewModel.ToolbarAnimator {
```
Changed the name of the class

teamagam3 commented on data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/AlertsDataRepository.java:
```diff
@@ -16,7 +16,7 @@
     @Inject
     public AlertsDataRepository() {
-        mReplayRepository = ReplayRepository.createReplayAll();
+        mReplayRepository = ReplayRepository.createReplayCount(1);
```
I understand why you changed this, but what if someone needs to get all of the alerts?

BarAgam replied:
That's right.

teamagam3 commented on data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/InformedAlertsDataRepository.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Inject
+    public InformedAlertsDataRepository() {
+        mLatestInformedDate = new Date(0);
```
Is there a Date.minValue() or something similar?

BarAgam replied:
Nope.

teamagam3 commented on data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java:
```diff
@@ -3,7 +3,7 @@
 
 public class Constants {
-    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev2.herokuapp.com";
+    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev1.herokuapp.com";
```
Revert changes

teamagam3 commented on domain/src/main/java/com/teamagam/gimelgimel/domain/alerts/UpdateLatestInformedAlertTimeInteractor.java:
```diff
@@ -0,0 +1,39 @@
+import rx.Observable;
+
+@AutoFactory
+public class UpdateLatestInformedAlertTimeInteractor extends BaseDataInteractor {
```
Don't we need a simpler class for this functionality?
It's not real "subsription request", we just want to update the InformedAlertsRepository.

Maybe I just don't understand the real meaning of this kind of class. Isn't it  just for long streams like messages streams?

BarAgam replied:
Basically we're using interactors for everything kind of action we want to make on our domain.
It's a good for several reasons, like uniformly manipulating domain from app and allows for logic to be implemented independently for each "use-case" (=interactors).

barmako committed:
CR fixes.
1. Alerts repository replays ALL alerts and not just the latest
2. A couple of name changes at BubbleAlertsSubcomponent and its ViewModel class.
